src/rust-crypto/CrossSigningIdentity.ts

```diff
@@ -66,18 +66,25 @@ export class CrossSigningIdentity {
         });
 
         if (olmDeviceHasKeys) {
-            if (!privateKeysInSecretStorage) {
+            if (!(await this.secretStorage.hasKey())) {
```

Comment:
this is the real fix in this PR: we need to check that secretStorage is set up before we try to export to it.